integration_test/test_all.py

```diff
@@ -59,12 +63,36 @@ def micro():
         instance.send('out', Message(0.1, data=result))
 
 
-def test_all(log_file_in_tmpdir):
+def check_profile_output(tmp_path):
+    conn = sqlite3.connect(tmp_path / 'performance.sqlite')
+    cur = conn.cursor()
+
+    for typ in ('SEND', 'RECEIVE_TRANSFER'):
+        cur.execute(
+                "SELECT * FROM all_events"
+                f"    WHERE instance = 'macro' AND type = '{typ}'")
```

Comment:
When you first mentioned this I felt a bit ashamed at having written this, as I have a degree in databases and SQL injection attacks really are security 101 there. Looking at it again though, it's just a test, it's all fixed strings in the end. So I'm not too worried anymore, but I'm changing it anyway if only to avoid setting a bad example and/or tripping some linter somewhere.